### PR TITLE
Add support for visiting tuples in walk() utility

### PR DESCRIPTION
**Pull request by antoine-morvan**
Hello, 

This PR addresses issue https://github.com/stfc/fparser/issues/367

I am not familiar with Python unit testing yet, thus did not add a test. Hope this is not too far off coding policies ^^'

Best.

### CODE REVIEW

**Comment by arporter** [comment]
Thanks @antoine-morvan, the fix looks good. Although codecov reports that it is covered by existing tests, I think it would be good to add a new, explicit test based on your reproducer. Would you like me to do that?

**Comment by antoine-morvan** [comment]
> Thanks @antoine-morvan, the fix looks good. Although codecov reports that it is covered by existing tests, I think it would be good to add a new, explicit test based on your reproducer. Would you like me to do that?

I will try to add one, that will be a good exercise before setting up tests in my code base :)

**Review by arporter — CHANGES_REQUESTED** [review]
Almost there now.
Only issue is the location of the new test (which is my fault because I didn't say where to put it).
Would you mind moving it into the `fparser/src/fparser/two/tests/utils/test_walk.py` file and removing the `test_issue367.py` file. (We inherited that way of naming test files when we took on fparser but have subsequently moved to structuring the tests to reflect the code structure.)

File: src/fparser/two/tests/utils/test_walk.py [hunk 1]
```diff
@@ -93,3 +95,26 @@ def test_walk_debug(capsys):
     stdout, _ = capsys.readouterr()
     assert stdout.startswith(8 * " " + "child type =")
     assert "Program" not in stdout
+
+def test_reproduce_issue367():
```
Comment (arporter) [inline src/fparser/two/tests/utils/test_walk.py:99]: Please could you rename this test to something like `test_walk_tuple` and add a brief doc string to the test.
This file also has a couple of Black formatting errors.
Once that's done we really should be ready to go :-)

**Review by arporter — CHANGES_REQUESTED** [review]
A couple of minor things need fixing and then we should be OK.

**Comment by antoine-morvan** [comment]
Should be better now :)

**Review by arporter — APPROVED** [review]
All requested changes have been made (thanks).
Black is happy and coverage is good.
Will proceed to merge.

**Pull request merged**